source/ranges.tex

```diff
@@ -12882,3 +12892,616 @@
+                     && @\libconcept{convertible_to}@<sentinel_t<V>, sentinel_t<@\exposid{Base}@>>;
+
+    constexpr iterator_t<@\exposid{Base}@> base() &&;
+    constexpr const iterator_t<@\exposid{Base}@>& base() const & noexcept;
+
+    constexpr decltype(auto) operator*() const { return *@\exposid{current_}@; }
+
+    constexpr @\exposid{iterator}@& operator++();
+    constexpr void operator++(int);
+    constexpr @\exposid{iterator}@ operator++(int) requires @\libconcept{forward_range}@<@\exposid{Base}@>;
+
+    constexpr @\exposid{iterator}@& operator--() requires @\libconcept{bidirectional_range}@<@\exposid{Base}@>;
+    constexpr @\exposid{iterator}@ operator--(int) requires @\libconcept{bidirectional_range}@<@\exposid{Base}@>;
+
+    constexpr @\exposid{iterator}@& operator+=(difference_type n) requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+    constexpr @\exposid{iterator}@& operator-=(difference_type n) requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+
+    constexpr decltype(auto) operator[](difference_type n) const
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>
+    { return *(*this + n); }
+
+    friend constexpr bool operator==(const @\exposid{iterator}@& x, default_sentinel_t);
+
+    friend constexpr bool operator==(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y)
+      requires @\libconcept{equality_comparable}@<iterator_t<@\exposid{Base}@>>;
+
+    friend constexpr bool operator<(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+
+    friend constexpr bool operator>(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+
+    friend constexpr bool operator<=(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+
+    friend constexpr bool operator>=(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+
+    friend constexpr auto operator<=>(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y)
+        requires @\libconcept{random_access_range}@<@\exposid{Base}@> && @\libconcept{three_way_comparable}@<iterator_t<@\exposid{Base}@>>;
+
+    friend constexpr @\exposid{iterator}@& operator+(const @\exposid{iterator}@& x, difference_type n)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+    friend constexpr @\exposid{iterator}@& operator+(difference_type n, const @\exposid{iterator}@& x)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
+    friend constexpr @\exposid{iterator}@& operator-(const @\exposid{iterator}@& x, difference_type n)
+      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
```

Comment:
Error in the ingress paper: All three of these should have a return type of `\exposid{iterator}`, _not_ `\exposid{iterator}&`. The return types are correct in the detailed specifications below.

Reply:
```suggestion
    friend constexpr @\exposid{iterator}@ operator+(const @\exposid{iterator}@& x, difference_type n)
      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
    friend constexpr @\exposid{iterator}@ operator+(difference_type n, const @\exposid{iterator}@& x)
      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
    friend constexpr @\exposid{iterator}@ operator-(const @\exposid{iterator}@& x, difference_type n)
      requires @\libconcept{random_access_range}@<@\exposid{Base}@>;
```